.github/workflows/deploy.yml

```diff
@@ -0,0 +1,39 @@
+name: Publish package to the Maven Central Repository
+on:
+  release:
+    types: [created]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '22' # or 11?
```

Comment:
My inclination would be to use a Java LTS version for compiling.

On the version itself, probably the biggest concern is about the compatiblity with earlier versions. Since the code uses the `java.nio.ByteBuffer`, using Java 11+ at compile time and Java 8 at runtime can result in exceptions of type `Exception in thread "main" java.lang.NoSuchMethodError: java.nio.ByteBuffer.position()Ljava/nio/ByteBuffer;`.

From experience, there are two routes:
- use the minimal supported Java LTS version for building the release artifacts, this is currently the strategy used by Bio-Formats
- use the `--release` command line option introduced in [JPE247](https://openjdk.org/jeps/247). For Maven, that should be the `maven.compiler.release` property - https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-release.html

Reply:
Thanks for the comment,
do you think compiling with Java 11 with `--release 8` will work?

Reply:
We have little experience with the `--release` flag but interested to see the outcome. In the context of the challenge, I'd vote for getting a first version of the library deployed to Maven Central. On our side with @melissalinkert, we can definitely ensure the Zarr 2 -> 3 conversion prototype is tested and executed on different Java environments including Java 8.